### PR TITLE
Make ISO creation more flexible

### PR DESCRIPTION
**Pull request by ghost**
- Make related configurations and variables clearer
- Add ISO creation instructions
- Leave only CentOS core packages in the Host OS ISO

### CODE REVIEW

**Comment by olavphibm** [comment]
start iso

File: BUILDING_AN_ISO.md [hunk 1]
```diff
@@ -0,0 +1,33 @@
...
+result directory, and may be installed using those
+[instructions](INSTALLING_PACKAGES#installation-using-iso-file).
+
+
+## Building a reduced ISO file
+
+To build a reduced ISO file, pass the groups you wish available in the ISO to
+the `--iso-repo-additional-packages-groups` option. The unavailable groups
+and environments will not show up during the installation, so other options do
+not need to be changed.
```
Comment (farosas) [inline BUILDING_AN_ISO.md]: What do you mean with this last phrase? It seems this is exposing implementation details in the doc.

File: BUILDING_AN_ISO.md [hunk 2]
```diff
@@ -0,0 +1,33 @@
...
+
+
+## Creating custom installable environments
+
+To create and add packages to instalable environments, modify the
```
Comment (farosas) [inline BUILDING_AN_ISO.md]: installable

File: README.md [hunk 3]
```diff
@@ -99,6 +99,13 @@ simplify using those repositories.
 
 To install them, follow the instructions [here](INSTALLING_PACKAGES.md).
 
+
+## Building an ISO
+
+To build an ISO with the built packages, follow those
+[instructions](BUILDING_AN_ISO).
```
Comment (farosas) [inline README.md:106]: Could you please remove this information from the CONTRIBUTING file? There's no need to duplicate instructions.

File: config.yaml [hunk 4]
```diff
@@ -61,19 +61,27 @@ build_iso:
...
-    iso_root_fs_packages_groups:
+    iso_repo_minimal_packages:
+        - "authconfig"
+        - "chrony"
+        - "grub2"
+    iso_repo_additional_packages_groups:
         - "base"
-        - "core"
         - "debugging"
         - "virtualization-hypervisor"
```
Comment (farosas) [inline config.yaml]: I thought you'd remove these...

File: lib/config.py [hunk 5]
```diff
@@ -119,19 +119,23 @@
...
-    ('--iso-root-fs-packages',):
-        dict(help='Packages that will be installed in the ISO root file '
-             'system, in addition to the specified groups.', nargs='*'),
+             'the same name, that will in turn contain the packages listed. '
+             'Those groups must be available in the ISO yum repository in '
+             'order to show up.'),
+    ('--iso-repo-minimal-packages-groups',):
+        dict(help='Packages groups that must be available in the ISO yum '
+             'repository and will always be installed in the target system.',
+             nargs='*'),
```
Comment (farosas) [inline lib/config.py]: In practice, what is the difference between adding packages/groups using these options vs. using the 'additional' ones?

**Review by farosas — DISMISSED** [review]

File: config.yaml (hunk 4 above)
Comment (ghost) [reply config.yaml]: I'll add another patch for that.

File: lib/config.py (hunk 5 above)
Comment (ghost) [reply lib/config.py]: In practice, none. I just splitted them to make it clear which groups can be changed and which will break the installation if removed. I didn't want to hide the required ones in a constant in the code.

File: README.md (hunk 3 above)
Comment (ghost) [reply README.md:106]: Yes, thanks for noticing that.

File: BUILDING_AN_ISO.md (hunk 1 above)
Comment (ghost) [reply BUILDING_AN_ISO.md]: It's supposed to mean that installable_environments whose groups are not listed here will not show up. Rephrasing.

**Comment by olavphibm** [comment]
start build

**Comment by olavphibm** [comment]
start build

**Comment by olavphibm** [comment]
start build

File: BUILDING_AN_ISO.md [hunk 6]
```diff
@@ -20,3 +20,12 @@ See `--help` for more options.
...
 ```
+
+
+## Creating custom installable environments
+
+To create and add packages to installable environments, modify the
+`installable_environments` option in the configuration. Each entry is an
+environment name that will show during installation in the "Software selection"
+screen and each item listed inside is a package that will be installed by that
+environemt.
```
Comment (farosas) [inline BUILDING_AN_ISO.md]: typo

**Review by farosas — APPROVED** [review]

**Pull request merged**